connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -118,11 +140,73 @@ public void report(ProcessingContext context) {
                     originalMessage.key(), originalMessage.value(), originalMessage.headers());
         }
 
+        if (connConfig.isDlqContextHeadersEnabled()) {
+            populateContextHeaders(producerRecord, context);
+        }
+
         this.kafkaProducer.send(producerRecord, (metadata, exception) -> {
             if (exception != null) {
                 log.error("Could not produce message to dead letter queue. topic=" + dlqTopicName, exception);
                 errorHandlingMetrics.recordDeadLetterQueueProduceFailed();
             }
         });
     }
+
+    // Visible for testing
+    void populateContextHeaders(ProducerRecord<byte[], byte[]> producerRecord, ProcessingContext context) {
+        String topic = "";
+        int partition = -1;
+        long offset = -1;
+        if (context.consumerRecord() != null) {
+            topic = context.consumerRecord().topic();
+            partition = context.consumerRecord().partition();
+            offset = context.consumerRecord().offset();
+        }
```

Comment:
Does it make more sense to only include the headers for the topic, partition, and offset if it is a consumer record? IIUC, right now we'll always include them.

Or, let me ask it this way: when would the context not have a consumer record? If there are conditions, should we should not include the headers for these under those conditions?